contracts/WooStakingVault.sol

```diff
@@ -76,60 +80,65 @@ contract WooStakingVault is ERC20, Ownable, Pausable {
-        treasury = _treasury;
+        require(initialStakedToken != address(0), 'WooStakingVault: initialStakedToken_ZERO_ADDR');
+        require(initialTreasury != address(0), 'WooStakingVault: initialTreasury_ZERO_ADDR');
+
+        stakedToken = IERC20(initialStakedToken);
+        treasury = initialTreasury;
     }
 
     /* ----- External Functions ----- */
 
-    function deposit(uint256 _amount) external whenNotPaused {
+    function deposit(uint256 amount) external whenNotPaused {
         uint256 balanceBefore = balance();
-        TransferHelper.safeTransferFrom(address(stakedToken), msg.sender, address(this), _amount);
+        TransferHelper.safeTransferFrom(address(stakedToken), msg.sender, address(this), amount);
         uint256 balanceAfter = balance();
-        _amount = balanceAfter.sub(balanceBefore);
+        amount = balanceAfter.sub(balanceBefore);
 
         uint256 xTotalSupply = totalSupply();
-        uint256 shares = xTotalSupply == 0 ? _amount : _amount.mul(xTotalSupply).div(balanceBefore);
+        uint256 shares = xTotalSupply == 0 ? amount : amount.mul(xTotalSupply).div(balanceBefore);
 
         // must be execute before _mint
-        _updateCostSharePrice(_amount, shares);
+        _updateCostSharePrice(amount, shares);
 
         _mint(msg.sender, shares);
 
-        emit Deposit(msg.sender, _amount, shares);
+        emit Deposit(msg.sender, amount, shares);
     }
 
-    function reserveWithdraw(uint256 _shares) external whenNotPaused {
-        uint256 currentReserveAmount = _shares.mul(getPricePerFullShare()).div(1e18); // calculate reserveAmount before _burn
+    function reserveWithdraw(uint256 shares) external whenNotPaused {
+        require(shares <= balanceOf(msg.sender), 'WooStakingVault: shares exceed balance');
+
+        uint256 currentReserveAmount = shares.mul(getPricePerFullShare()).div(1e18); // calculate reserveAmount before _burn
         uint256 poolBalance = balance();
         if (poolBalance < currentReserveAmount) {
             // incase reserve amount exceeds pool balance
             currentReserveAmount = poolBalance;
         }
-        _burn(msg.sender, _shares); // _burn will check the balance of user's shares enough or not
+        _burn(msg.sender, shares); // _burn will check the balance of user's shares enough or not
 
         totalReserveAmount = totalReserveAmount.add(currentReserveAmount);
 
         UserInfo storage user = userInfo[msg.sender];
         user.reserveAmount = user.reserveAmount.add(currentReserveAmount);
         user.lastReserveWithdrawTime = block.timestamp;
 
-        emit ReserveWithdraw(msg.sender, currentReserveAmount, _shares);
+        emit ReserveWithdraw(msg.sender, currentReserveAmount, shares);
     }
 
     function withdraw() external whenNotPaused {
         UserInfo storage user = userInfo[msg.sender];
         uint256 withdrawAmount = user.reserveAmount;
+        uint256 currentWithdrawFee = 0;
```

Comment:
Seems like "current" prefix is not needed ?

Reply:
there's a status variable name withdrawFee, therefore here current is needed in case duplicate